### PR TITLE
HTTPS everywhere

### PR DESCRIPTION
**Pull request by enolfc**
# Summary

Add instructions to create a cert and enable it in the Jupyter + DataHub VM

---

**Related issue :**

### CODE REVIEW

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md [hunk 1]
```diff
@@ -48,7 +48,60 @@ Once your instance is ready,
...
-### Step 2: Get your token for the Jupyter server
+### Step 2: Get a hostname and certificate for your VM
+
+Your VM is ready to be accessible, but runs a plain HTTP server, which is not
+secure enough. If you try to connect with your browser to your VM, you will get
+a message as shown in the screenshot below:
+
+![no-https](no-https-datahub-vm.png)
+
+You can enable HTTPS to encrypt requests and responses, thus making your VM
```
Comment (gwarf) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:59]:
```suggestion
You must enable HTTPS to encrypt requests and responses, thus making your VM
```

Or at least should.

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md [hunk 2]
```diff
@@ -48,7 +48,60 @@ Once your instance is ready,
...
+safer and more secure.
+
+First, you need a valid name for your VM. You can use the
```
Comment (gwarf) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:62]:
```suggestion
Firstly, you need a valid name for your VM. You can use the
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md [hunk 3]
```diff
@@ -48,7 +48,60 @@ Once your instance is ready,
...
+[FedCloud Dynamic DNS](https://nsupdate.fedcloud.eu) to create a name. See
+[Dynamic DNS docs](../../compute/cloud-compute/dynamic-dns/) for more
+information on the service. Once you have your name ready, assign it your VM's
+IP.
+
+Second, you need to get a certificate to enable HTTPS. The VM has
```
Comment (gwarf) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:68]:
```suggestion
Secondly, you need to get a certificate to enable HTTPS. The VM has
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md [hunk 4]
```diff
@@ -48,7 +48,60 @@ Once your instance is ready,
...
+```
+
+Finally, open your browser and go to `https://<your registered name>/` to see
+Jupyter started. Follow next steps for getting the credentials to access the
+service.
+
+{{% alert title="Enabling insecure access" color="Danger" %}} If you really need
+to use HTTP (e.g. your VM is running without inbound connectivity and cannot
+create a certificate for it), you can disable the error shown by default in the
+nginx configuration.
```
Comment (gwarf) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:86]: I would expect that if someone is trying to access the VM on http, there is some inbound connectivity, maybe we should replace by something like public or saying it's used "internally"?
Is this kind of setup something that is really used?

**Review by gwarf — CHANGES_REQUESTED** [review]
That's great, thanks!
I added a few suggestions/comments.

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md [hunk 5]
```diff
@@ -48,7 +48,60 @@ Once your instance is ready,
...
+# now request the certificate
+$ sudo certbot --nginx -d <your registered name> -m <your email>
+```
+
+Finally, open your browser and go to `https://<your registered name>/` to see
+Jupyter started. Follow next steps for getting the credentials to access the
+service.
+
+{{% alert title="Enabling insecure access" color="Danger" %}} If you really need
+to use HTTP (e.g. your VM is running without inbound connectivity and cannot
```
Comment (enolfc) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:84]:
```suggestion
to use HTTP (e.g. your VM is not accessible publicly and you cannot
```

Comment (enolfc) [reply content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:84]: is this better?

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md (hunk 4 above)
Comment (gwarf) [reply content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md]: OK, it's a fair use case.

**Review by gwarf — APPROVED** [review]
LGTM

**Comment by enolfc** [comment]
> @enolfc, now all is green, should we wait or merge?

Let's merge

**Pull request merged**